### DISCUSSION TITLE
Why does IS4 JWT token expire sooner than expected?

### DISCUSSION THREAD

**Discussion by aspramsh** [ThreeMammals/Ocelot#1880] [Q&A] [answered]
I am using identity server tokens in ocelot gateway. Everything works fine. However I have a month lifetime for my access tokens in my testing environment, but after generating a token usually in 2 days of using it I get 401 unauthorized. Here is an example of my token that is regarded as expired: `eyJhbGciOiJSUzI1NiIsImtpZCI6IjFGREFDOTI0RDFCQTU5N0ZDODJERjIwMjg2RTAxMzk2IiwidHlwIjoiYXQrand0In0.eyJuYmYiOjE2Mzc3MzQ5MjcsImV4cCI6MTY0MDMyNjkyNywiaXNzIjoiaHR0cDovL3Vzcm1uZ210IiwiYXVkIjoiZW1iZWRkZWRfZmluYW5jZV9hcGkiLCJjbGllbnRfaWQiOiJuaXN5YV9jbGllbnQiLCJpYXQiOjE2Mzc3MzQ5MjcsInNjb3BlIjpbImVtYmVkZGVkX2ZpbmFuY2UiXX0.Kl636wBNiAO3i8QrAZos4J3DrSd8-hyCGJ8w6VstdhZvZCpt3NI1WqGLc5kxD9G61cuUhu-gDq4DKIV4A2e8IT1R1ljaAbQDZXOvcySgR0WbpHN4kHz3wJZQpYfe25TCuI_28RF6FRaRmDOZVZ1NeGypGGazKOSVnrCVxXApRwe2DbyzFA6oteW1jnCGNFsJ02f4aMA4TQotla6U-sFmbX_Km1oZDHQFqwEtEE-8RDY7tMYnYFv9HdKfP9yGPCp8BEjQEMXSln1xRWXOeShQSPvSCLI5CSUU17UlMxGtodXFXYkmfl-n4__P3AQ6eGtWY-7djQwKZcNz_Jv2k8WAvQ`

You can see in https://jwt.io/ that it is perfectly valid. And here is a newly generated token that works perfectly:
`eyJhbGciOiJSUzI1NiIsImtpZCI6IjZEMjU0Qzc2RThDOUIwOEZDN0I0N0YyQTcyQzhDNEQ5IiwidHlwIjoiYXQrand0In0.eyJuYmYiOjE2Mzc5MTEwNjAsImV4cCI6MTY0MDUwMzA2MCwiaXNzIjoiaHR0cDovL3Vzcm1uZ210IiwiYXVkIjoiZW1iZWRkZWRfZmluYW5jZV9hcGkiLCJjbGllbnRfaWQiOiJuaXN5YV9jbGllbnQiLCJpYXQiOjE2Mzc5MTEwNjAsInNjb3BlIjpbImVtYmVkZGVkX2ZpbmFuY2UiXX0.TDRnEKNKKvkeNV3kVJm-AYLQqwhx0C-nkTowFi5L8hXTrWLebqLlYqV-LoyQbshsHfrcMPxd7EghNgnh41pOtOTOOGQI5G8-9aPB2ldAAsh3OtfqmAcZg5iI_U9lP2jMBEBq0A2IhtXSRmx3Ky2qwgsa3U9s7m0fZw-0wBKppG4vH9gYAeFhtwit7fpUjwaO9fzMdCQn5y17djO6VvzPXOCIaQ-VHWemXwRvhcnpIw3pi66H4zpH8ExMFyNmhRRnxcDG4QlMTXesO-DYgTzfxKoT4mNHiV_jZFrjwo6oWpK5n8Ja2gkT3gDnFHfxYOEnPfYygNytSI2Fvg5KtY-Xqw`

I just can't understand what is the problem. Any help would be highly appreciated.

**Comment by abdulmoizhussain** [comment, marked as answer]
Inside your Identity Server project, if you are using `.AddDeveloperSigningCredential()`,
then try to use `.AddSigningCredential()` with your own certificate.
